web/src/elements/wizard/Wizard.ts

```diff
@@ -20,91 +21,149 @@ export const ApplyActionsSlot = "apply-actions";
+
+    @property({ attribute: false })
+    state: { [key: string]: unknown } = {};
+
+    //#endregion
+
+    //#region State
+
+    /**
+     * Memoized step tag names.
+     */
     @state()
     _steps: string[] = [];
 
+    /**
+     * Step tag names present in the wizard.
+     */
     get steps(): string[] {
         return this._steps;
     }
 
-    set steps(steps: string[]) {
-        const addApplyActionsSlot = this.steps.includes(ApplyActionsSlot);
-        this._steps = steps;
+    set steps(nextSteps: string[]) {
+        const addApplyActionsSlot = this._steps.includes(ApplyActionsSlot);
+
+        this._steps = nextSteps;
+
         if (addApplyActionsSlot) {
             this.steps.push(ApplyActionsSlot);
         }
-        this.steps.forEach((step) => {
-            const exists = this.querySelector(`[slot=${step}]`) !== null;
-            if (!exists) {
-                const el = document.createElement(step);
-                el.slot = step;
-                el.dataset["wizardmanaged"] = "true";
-                this.appendChild(el);
-            }
-        });
+
+        for (const step of this._steps) {
+            const existingStepElement = this.getStepElementByName(step);
+
+            if (existingStepElement) continue;
+
+            const stepElement = document.createElement(step);
+
+            stepElement.slot = step;
+            stepElement.dataset["wizardmanaged"] = "true";
+
+            this.appendChild(stepElement);
+        }
+
         this.requestUpdate();
     }
 
+    /**
+     * Initial steps to reset to.
+     */
     _initialSteps: string[] = [];
 
-    @property({ attribute: false })
-    actions: WizardAction[] = [];
-
     @state()
-    _currentStep?: WizardPage;
+    _activeStep?: WizardPage;
 
-    set currentStep(value: WizardPage | undefined) {
-        this._currentStep = value;
-        if (this._currentStep) {
-            this._currentStep.activeCallback();
-            this._currentStep.requestUpdate();
-        }
+    set activeStepElement(nextActiveStepElement: WizardPage | undefined) {
+        this._activeStep = nextActiveStepElement;
+
+        if (!this._activeStep) return;
+
+        this._activeStep.activeCallback();
+        this._activeStep.requestUpdate();
     }
 
-    get currentStep(): WizardPage | undefined {
-        return this._currentStep;
+    /**
+     * The active step element being displayed.
+     */
+    get activeStepElement(): WizardPage | undefined {
+        return this._activeStep;
     }
 
-    @property({ attribute: false })
-    finalHandler: () => Promise<void> = () => {
-        return Promise.resolve();
-    };
+    getStepElementByIndex(stepIndex: number): WizardPage | null {
+        const stepName = this._steps[stepIndex];
 
-    @property({ attribute: false })
-    state: { [key: string]: unknown } = {};
+        return this.querySelector<WizardPage>(`[slot=${stepName}]`);
+    }
+
+    getStepElementByName(stepName: string): WizardPage | null {
```

Comment:
I suspect these could be ref'd at some point but if they're like React refs, dealing with live element arrays is tricky. Another idea was to use a Map, but I could see it getting out of sync with the DOM.